src/Toml/Bi/Code.hs

```diff
@@ -88,7 +89,12 @@ type TomlCodec a = BiCodec Env St a
 decode :: TomlCodec a -> Text -> Either DecodeException a
 decode codec text = do
     toml <- first ParseError (parse text)
-    runReader (runExceptT $ codecRead codec) toml
+    decodeToml codec toml
+
+-- | Convert toml into user data type.
+decodeToml :: TomlCodec a -> TOML -> Either DecodeException a
+decodeToml codec =
+    runReader (runExceptT $ codecRead codec)
```

Comment:
I think that this function can be implemented in single line, no need to have extra line break